wagtail/models/__init__.py

```diff
@@ -1140,7 +1140,9 @@ class Page(AbstractPage, index.Indexed, ClusterableModel, metaclass=PageBase):
         on_delete=models.SET(get_default_page_content_type),
     )
     content_type.wagtail_reference_index_ignore = True
-    url_path = models.TextField(verbose_name=_("URL path"), blank=True, editable=False)
+    url_path = models.TextField(
+        verbose_name=_("URL path"), blank=True, editable=False, max_length=2000
```

Comment:
Do we actually need to change this? Previously this was a TextField with no max length - it seems to me that adding a new limit has no benefit, but could potentially break existing sites (if they had >=2000 character paths, and didn't experience any problems with this because they weren't using redirects).

Reply:
@gasman there was quite a discussion about it in the issue that led to this PR #11053

Reply:
@gasman Sites with paths longer than 2000 characters are already probably broken because browsers don't support paths that long.

That being said, I think we could change this to a `CharField` instead of being a `TextField`, although it will be technically the same thing under the hood of most databases.

Reply:
@BertrandBordage I'd rather not make any changes to the core Page model if it doesn't actually contribute towards fixing #11053. I don't know how likely it is that someone will run into the 2000-character limit, or how catastrophic it will be to their database if they do, but if there's an option that saves me from having to think about that possibility, I'll take it.

Reply:
@gasman Then, as discussed in #11053, do you think we should therefore change `Redirect.old_path` to a `TextField` and make it use a `TextInput` instead of `TextArea`? That would be the only way to make `Redirect.old_path` consistent with `Page.url_path` without changing `Page.url_path`.

Reply:
https://stackoverflow.com/a/417184/201316 suggests the safe limit is 2000 chars cross-browser, but that varies wildly

Reply:
@BertrandBordage If you feel that it's worth jumping through the extra hoop to make the definitions exactly consistent, then yes - but I would be equally happy with defining the redirect path as `CharField(max_length=2000)` and leaving Page unchanged.

We're all working under the assumption that no Wagtail user is reasonably going to hit the 2000 character limit. If that assumption is true, then the two definitions (`CharField(max_length=2000)` and `TextField()`) _are_ consistent for all practical purposes, and the difference between them is academic. On the other hand, if that assumption turns out to be false, I'd like the damage from that faulty assumption to be contained within the wagtail.contrib.redirects app, rather than causing new problems on the core Page model.

Reply:
@gasman Sounds good!

Reply:
@gasman @zerolab @BertrandBordage thanks for the this. I will make adjustments to the PR accordingly.